### PR TITLE
Conditional Templates

### LINKED ISSUE #2741: Enable optional fields to show/hide in UI
As templates become more advanced, we need to support a more dynamic form template, implementing support for `allOf`, `if/else` in the template json. 

Some UI-specific changes will need to be made as the jrsf library is pretty fussy about rendering when there are null fields in the template.

### PR DESCRIPTION
**Pull request by damoodamoo**
# Resolves #2741 

- Implements the `allOf` feature of JSON schema to allow conditional fields + conditional schema validation
- "auto_create" string has gone in favour of an enum (dropdown), which shows appropriate follow-up fields
- all workspace templates updated to use conditional schema to streamline the form
- Airlock review VM configuration - nested properties, conditionally required
- UI code to massage the data payload sent in for create or update of a resource. 

image

### CODE REVIEW

File: templates/core/terraform/variables.tf [hunk 1]
```diff
@@ -61,7 +61,7 @@ variable "terraform_state_container_name" {
 
 variable "resource_processor_number_processes_per_instance" {
   type        = string
-  default     = "2"
+  default     = "5"
```
Comment (tanya-borisova) [inline templates/core/terraform/variables.tf:64]: This definitely needs to be taken out of this PR

**Review by tanya-borisova — CHANGES_REQUESTED** [review]
I think this PR lends itself naturally to being split into several smaller PRs: 

- Increasing the number of processes in resource processor (AFAIU it is not related to this pull request?)
- Endpoints for getting a template with a specific version
- Adding support for AllOf in the templates including the changes on UI necessary
- Replacing auto_create with Automatic and manual auth_type (am I right that it is just refactoring and does not depend on the rest of the PR?)
- Finally, changing the workspace templates to take advantage of the introduced AllOf feature for Airlock configuration

As it stands, I would be reluctant to approve this PR as there are too many changes. It is likely that I and other reviewers will miss something while reviewing.
It would also be good to call out these changes as separate items in CHANGELOG.

File: api_app/api/routes/airlock.py [hunk 2]
```diff
@@ -130,13 +130,15 @@ async def create_review_user_resource(
         # Getting the review configuration from the airlock request's workspace properties
         if airlock_request.type == AirlockRequestType.Import:
             config = workspace.properties["airlock_review_config"]["import"]
-            workspace_id = config["workspace_id"]
+            workspace_id = config["import_vm_workspace_id"]
+            workspace_service_id = config["import_vm_workspace_service_id"]
+            user_resource_template_name = config["import_vm_user_resource_template_name"]
         else:
             assert airlock_request.type == AirlockRequestType.Export
             config = workspace.properties["airlock_review_config"]["export"]
```
[the remark this answers is not in the document]
Comment (damoodamoo) [reply api_app/api/routes/airlock.py:138]: the nested structure is still there, so there are parent objects for `import` and `export` still.

File: api_app/api/routes/airlock.py [hunk 3]
```diff
@@ -130,13 +130,15 @@ async def create_review_user_resource(
         # Getting the review configuration from the airlock request's workspace properties
         if airlock_request.type == AirlockRequestType.Import:
             config = workspace.properties["airlock_review_config"]["import"]
```
[the remark this answers is not in the document]
Comment (damoodamoo) [reply api_app/api/routes/airlock.py:132]: yep - as mentioned below, the structure is still:
```
properties: {
   ... 
   {
      "airlock_review_config": { 
          "import": {
              ...
          }, "export": { } 
   }
}
```

**Comment by damoodamoo** [comment]
Closing as env is messed up. Creating new.

**Pull request closed without merging**